src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1798,7 +1793,27 @@ trait Contexts { self: Analyzer =>
       result.filter(sym =>
         if (isRootImport) !definitions.isUnimportableUnlessRenamed(sym)
         else definitions.isImportable(sym)
-      )
+      ) match {
+        case filtered: NoSymbol => (null, filtered)
+        case _                  => (current, result)
+      }
+    }
+
+    private def selectorString(s: ImportSelector): String = {
+      if (s.isWildcard) "_"
+      else if (s.isRename) s.name + " => " + s.rename
+      else "" + s.name
+    }
+
+    /** Optionally record that a selector was used to import the given symbol. */
+    def recordUsage(sel: ImportSelector, result: Symbol): Unit = {
+      debuglog(s"In $this at ${ pos.source.file.name }:${ posOf(sel).line }, selector '${ selectorString(sel)
```

Comment:
I was focused on nontrivial rebase; I'll take a second look.

Reply:
@lrytz Actually this will need a follow-up. The point of the "record" flag was to distinguish look-ups of implicits, which are not yet usages. Or wait, I guess it's OK, implicits collection uses `importedSymbol` which ignores the guilty selector and records nothing; `SymbolLookup.apply` has `lookupImport` that tracks the selector which is then recorded. All the words are permutations of the same words, and all the types are combinations of `Name` and `Symbol`. I'm also suspicious of the `requireExplicit` flag.

Reply:
It was a rebase error to move `def selectorString` back outside `recordUsage`.